File: Selection Sort/selection.cpp
#include <iostream>
using namespace std;

void sort(double * A, int n){
	for(int i=0; i<n-1; i++){
		int minInd = i;
		for(int j=i+1; j<n; j++)
			if(A[j] < A[minInd]) minInd=j; //in ascending order. replace < with > for descending (change minInd to maxInd too)
		double t = A[i];
		A[i] = A[minInd];
		A[minInd] = t;
	}
}

int main(){
	int n;
	cin >> n;
	double *A = new double[n];
	for(int i=0; i<n; cin >> A[i++]);
	sort(A,n);
	for(int i=0; i<n; cout <<A[i++]<<" ");
}

File: Selection Sort/Selection.java
import java.util.*;
class Selection{
	
	public static void main(String args[]){
		ArrayList<Double> A = new ArrayList<Double>();
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		for(int i=0; i<n; A.add(sc.nextDouble()),i++);
		sort(A);
		for(int i=0; i<n; System.out.print(A.get(i++) +" "));
	}

	static void sort(ArrayList<Double> A){
		for(int i=0; i<A.size()-1; i++){
			int minInd = i;
			for(int j=i+1; j<A.size(); j++)
				if(A.get(j) < A.get(minInd)) minInd=j; //in ascending order. replace < with > for descending (change minInd to maxInd too)
			double t = A.get(i);
			A.set(i, A.get(minInd));
			A.set(minInd, t);
		}
	}
}